### DISCUSSION TITLE
Tilemap should have separate Light Mask for each layer

### DISCUSSION THREAD

**Discussion by Perpetuum1101** [godotengine/godot-proposals#6911] [2D]
This functionality is available in Unity. I think it would be nice to see the floor illuminated while the walls are not.

**Comment by Ralgondo** [comment]
Yea I was trying to figure out a way to make this happen as well...

**Comment by WilhelmT** [comment]
I was looking for exactly this as well! 😅

**Comment by AThousandShips** [comment]
Since `TileMap` is deprecated it's unlikely to be added, is it not covered by `TileMapLayer`?

**Reply by WilhelmT** [reply]
It's not fixed by TileMapLayer, since the light mask is set for the TileSet, which is (still) used by all TileMapLayer

**Reply by AThousandShips** [reply]
You just set it for the tile no? Visibility isn't per `TileSet`?

**Reply by WilhelmT** [reply]
Yes, I meant to say that it's set per tile, sorry for the confusion.

Setting it per tile means that it gets applied to all TileMapLayers since tiles are modified through on the level of a TileSet, which is shared for all TileMapLayers. Any change to tiles gets applied to all TIleMapLayers, and hence they cannot have separate light masks.

**Reply by AThousandShips** [reply]
That's separate from the original discussion, so that's something you should make a new proposal or discussion for, the walls would be a separate tile from the floor so they can have a different light management, having the *same* tile behave differently in different layers is pretty different (and you can already do it with tile alternatives)
